File: DSA/Pranvkumar_590011587/Day34_Question3.py
# 3. Check if Binary Representation is a Palindrome
def is_binary_palindrome(n):
    num_bits = n.bit_length()
    for i in range(num_bits // 2):
        left_bit = (n >> (num_bits - 1 - i)) & 1
        right_bit = (n >> i) & 1
        if left_bit != right_bit:
            return "No"
    return "Yes"

# --- Test Cases ---
if __name__ == "__main__":
    # Binary Palindrome Check
    print("\nBinary Palindrome Check:")
    print("Input: 9 -> Output:", is_binary_palindrome(9))    
    print("Input: 10 -> Output:", is_binary_palindrome(10))

File: DSA/Ayush_Mishra_590014844/Day_34_Question_3.c
#include <stdio.h>

int main() {
    unsigned int n;
    printf("Enter an integer to check if its binary representation is a palindrome: ");
    scanf("%u", &n);

    int left = sizeof(n) * 8 - 1;
    int right = 0;
    int is_palindrome = 1;

    while (left > right) {
        int left_bit = (n >> left) & 1;
        int right_bit = (n >> right) & 1;
        if (left_bit != right_bit) {
            is_palindrome = 0;
            break;
        }
        left--;
        right++;
    }

    if (is_palindrome)
        printf("Yes, the binary representation is a palindrome.\n");
    else
        printf("No, the binary representation is not a palindrome.\n");

    return 0;
}
